### DISCUSSION TITLE
Trouble of pore.labels

### DISCUSSION THREAD

**Discussion by Xknightfyy** [PMEAL/porespy#772] [Q&A] [answered]
Dear Professor Jeff Gostick,
 
I'm a beginner and I can't get labels like pore.left when I use poresspy to extract pore network models and import them into openpnm. Can you tell me how to add pore.labels? It would be nice to have sample code.
<img width="452" alt="1670638598745" src="https://user-images.githubusercontent.com/120148734/206823973-7378210f-ed90-4b27-be82-bdb2f634402f.png">

**Comment by Xknightfyy** [comment, marked as answer]
``` python
import imageio
import numpy as np
import porespy as ps
import openpnm as op
import scipy.ndimage as spim
import matplotlib.pyplot as plt
from porespy.filters import find_peaks, trim_saddle_points, trim_nearby_peaks
from porespy.tools import randomize_colors
from skimage.segmentation import watershed

ps.settings.tqdm['disable'] = True
ps.visualization.set_mpl_style()
np.random.seed(10)

#
im = ps.generators.cylinders(shape=[200, 200, 200], r=5, ncylinders=200)
fig, ax = plt.subplots()
ax.imshow(ps.visualization.show_planes(im), cmap=plt.cm.bone);

sigma = 0.4
dt = spim.distance_transform_edt(input=im)
dt1 = spim.gaussian_filter(input=dt, sigma=sigma)
peaks = find_peaks(dt=dt)

print('Initial number of peaks: ', spim.label(peaks)[1])
peaks = trim_saddle_points(peaks=peaks, dt=dt1)
print('Peaks after trimming saddle points: ', spim.label(peaks)[1])
peaks = trim_nearby_peaks(peaks=peaks, dt=dt)
peaks, N = spim.label(peaks)
print('Peaks after trimming nearby peaks: ', N)

#
regions = watershed(image=-dt, markers=peaks, mask=dt > 0)
regions = randomize_colors(regions)

fig, ax = plt.subplots()
ax.imshow((regions*im)[:, :, 100], cmap=plt.cm.nipy_spectral);

net = ps.networks.regions_to_network(regions*im, voxel_size=1)
pn = op.io.network_from_porespy(net)
pn.regenerate_models()
ax = op.visualization.plot_connections(pn)
ax = op.visualization.plot_coordinates(pn, ax=ax)
print(pn)

#snow_output = ps.networks.snow(
#    im,
#    voxel_size=1,
#    boundary_faces=['left','right','front','back'],
#    marching_cubes_area=False
#)
#
del net.models['pore.diameter']
del net.models['throat.diameter']
del net.models['throat.coords']

net['pore.diameter'] = net['pore.inscribed_diameter']
net['throat.diameter'] = net['throat.inscribed_diameter']
net['throat.coords'] = net['throat.global_peak']
```

**Reply by jgostick** [reply]
I'm actually not sure what your question is...I think you are wondering why the label `'pore.left'` is missing from the network after you open it in OpenPNM?  In v2 of porespy we switched to using `'pore.xmin'` and `'pore.xmax'` which is more explicit about which face it refers to.  

However, in the above code you are doing the extraction yourself using each of the individual steps.  This is ok, but you need to then add labels yourself too.  You can do this by basically checking which pores are on or near the left (i.e. xmin) face, with something like:

`Ps = pn['pore.coords'][:, 0] < (pn['pore.coords'][:, 0].min()*1.1)`

This will find all pores which have an x-coordinate equal to the 10% of the minimum of all x coordinates.

You can then create the label with:

``` python
pn['pore.left'] = False
pn['pore.left'][Ps] = True
```

**Reply by Xknightfyy** [reply]
Thanks for the answer, I have now solved this issue, thank you very much
